src/resources/ApiKeys/ApiKeysInterfaces.ts

```diff
@@ -109,9 +109,9 @@ interface CommerceConfigurationModel {
 }
 
 interface SearchConfigurationModel {
-    apiKeyQueryAuthentication: QueryAuthenticationModel[];
-    enforcedQueryPipelineConfiguration: EnforceQueryPipelineConfigurationModel;
-    impersonationRestriction: ImpersonationRestrictionsModel;
+    apiKeyQueryAuthentication?: QueryAuthenticationModel[];
+    enforcedQueryPipelineConfiguration?: EnforceQueryPipelineConfigurationModel;
+    impersonationRestriction?: ImpersonationRestrictionsModel;

```

Comment:
Which of the following does the API accept: empty arrays, null, undefined?

Reply:
Not sure, I know that undefined is accepted for `apiKeyQueryAuthentication` and `impersonationRestriction` (I can create/update/get an API key with only `enforcedQueryPipelineConfiguration` set).

Reply:
Ok:
- So, the API with `enforcedQueryPipelineConfiguration`, `apiKeyQueryAuthentication` and `impersonationRestriction` undefined is failing?
- What if `apiKeyQueryAuthentication` or `impersonationRestriction` is defined and `enforceQueryPipelineConfiguration` is undefined?


The thing here is that by making all props undefined, you say "it's fine to send a SearchConfigurationModel with all its key undefined".
That might be the case, I do not know, but as a reviewer, I outta ask you to check ;)
If that's not the case, then the interface must be tweaked to reflect the actual acceptance of the API IMHO, otherwise, it means that the platform-client could induce a user in error, which is meh :|

Reply:
Unfortunately,I didn't implement this API (so I cannot answer your question right away) and I need this information to resolve a bug. 

To speed up the process, would it be better to actually remove the parameters (or comment them with an explanation) `apiKeyQueryAuthentication` and `impersonationRestriction`, set the `enforcedQueryPipelineConfiguration` as mandatory and the actual developers that worked on that section update the information correctly? What is blocking me at the moment is the fact that `search` is mandatory, and it should not.